File: driver/python/task.py
from message import * 
from pprint import pprint

class Task:
	
	def __init__(self, agent, function_name, workload):
		self.agent = agent
		self.function_name = function_name
		self.workload = workload
		self.progress = 0
		self.result = None
		self.id = None
		self.status = 'created'
		self.connection_id = None
		
	def updateInfo(self, message):
		self.id = message.body['jobid']
		
		#pprint((vars(message)))
		if 'connection_id' in message.header:
			self.connection_id = message.header['connection_id']
		
		if message.getCode() == Message.WORK_STATUS:
			self.status = message.body['status']
			self.progress = message.body['progress']
		elif message.getCode() == Message.WORK_DONE:
			self.status = 'finished'
			self.result = message.body['result']
			self.progress = 100
		
		
	def setStatus(progress):
		self.progress = progress
		if self.agent.type == 'worker':
			self.agent.notifyWorkStatus(self)

File: driver/php/Task.php
<?php

namespace gtsd;

class Task
{
    public $id;
    public $progress = 0;
    public $status = 'created';
    public $function_name;
    public $workload;
    public $result;
    public $connId;
    public $agent;
    
    
    public function __construct($agent, $function_name, $workload)
    {
        $this->agent = $agent;
        $this->function_name = $function_name;
        $this->workload = $workload;
    }
    
    public function updateInfo(Message $message)
    {
        $this->id = $message->getBodyData('jobid');
        
        switch($message->getCode())
        {
            case Message::WORK_STATUS:
                $this->status = $message->getBodyData('status');
                $this->progress = $message->getBodyData('progress');
                break;
            
            case Message::WORK_DONE:
                $this->status = 'finished';
                $this->result = $message->getBodyData('result');
                $this->progress = 100;
                break;
        }
    }
    
    public function setStatus($progress)
    {
        $this->progress = $progress;
        if($this->agent instanceof Worker)
        {
            $this->agent->notifyWorkStatus($this);
        }
    }
}

?>
